firefox-ios/Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift

```diff
@@ -27,33 +30,68 @@ class EnhancedTrackingProtectionCoordinator: BaseCoordinator,
         let url = tab?.url ?? URL(fileURLWithPath: "")
         let displayTitle = tab?.displayTitle ?? ""
         let contentBlockerStatus = tab?.contentBlocker?.status ?? .blocking
+        let contentBlockerStats = tab?.contentBlocker?.stats
         let connectionSecure = tab?.webView?.hasOnlySecureContent ?? true
-        let etpViewModel = EnhancedTrackingProtectionMenuVM(
+        let etpViewModel = TrackingProtectionModel(
             url: url,
             displayTitle: displayTitle,
             connectionSecure: connectionSecure,
             globalETPIsEnabled: FirefoxTabContentBlocker.isTrackingProtectionEnabled(prefs: profile.prefs),
-            contentBlockerStatus: contentBlockerStatus)
+            contentBlockerStatus: contentBlockerStatus,
+            contentBlockerStats: contentBlockerStats
+        )
 
-        self.enhancedTrackingProtectionMenuVC = EnhancedTrackingProtectionMenuVC(viewModel: etpViewModel,
+        self.enhancedTrackingProtectionMenuVC = TrackingProtectionViewController(viewModel: etpViewModel,
                                                                                  windowUUID: tabManager.windowUUID)
+        let oldEtpViewModel = EnhancedTrackingProtectionMenuVM(
+            url: url,
+            displayTitle: displayTitle,
+            connectionSecure: connectionSecure,
+            globalETPIsEnabled: FirefoxTabContentBlocker.isTrackingProtectionEnabled(prefs: profile.prefs),
+            contentBlockerStatus: contentBlockerStatus
+        )
+
+        self.legacyEnhancedTrackingProtectionMenuVC = EnhancedTrackingProtectionMenuVC(viewModel: oldEtpViewModel,
+                                                                                       windowUUID: tabManager.windowUUID)
         self.profile = profile
         self.tabManager = tabManager
         super.init(router: router)
         enhancedTrackingProtectionMenuVC.enhancedTrackingProtectionMenuDelegate = self
     }
 
     func start(sourceView: UIView) {
-        if UIDevice.current.userInterfaceIdiom == .phone {
-            enhancedTrackingProtectionMenuVC.modalPresentationStyle = .custom
-            enhancedTrackingProtectionMenuVC.transitioningDelegate = self
+        let trackingProtectionRefactorStatus =
+        featureFlags.isFeatureEnabled(.trackingProtectionRefactor, checking: .buildOnly)
+        if trackingProtectionRefactorStatus {
+            if UIDevice.current.userInterfaceIdiom == .phone {
+                if let sheetPresentationController = enhancedTrackingProtectionMenuVC.sheetPresentationController {
+                    sheetPresentationController.detents = [.medium(), .large()]
+                    sheetPresentationController.prefersScrollingExpandsWhenScrolledToEdge = true
+                    sheetPresentationController.preferredCornerRadius = 12
+                }
+                router.present(enhancedTrackingProtectionMenuVC, animated: true, completion: nil)
+            } else {
```

Comment:
how does this look like in multitasking? do we want to use `shouldUseiPad` check instead?

Reply:
Not sure about this comment here but line 72 will be executed only for .phone cases and not for iPads.

Reply:
As I tested, it is working well on multitasking / split view mode for iPad.

Reply:
okay, thanks for checking!